static/app/views/issueDetails/groupEventAttachments/groupEventAttachments.tsx

```diff
@@ -25,167 +27,143 @@ import GroupEventAttachmentsFilter, {
 import GroupEventAttachmentsTable from './groupEventAttachmentsTable';
 import {ScreenshotCard} from './screenshotCard';
 
-type Props = {
+type GroupEventAttachmentsProps = {
   project: Project;
-} & WithRouterProps<{groupId: string; orgId: string}> &
-  DeprecatedAsyncComponent['props'];
+};
 
 enum EventAttachmentFilter {
   ALL = 'all',
   CRASH_REPORTS = 'onlyCrash',
   SCREENSHOTS = 'screenshot',
 }
 
-type State = {
-  deletedAttachments: string[];
-  eventAttachments?: IssueAttachment[];
-} & DeprecatedAsyncComponent['state'];
-
 export const MAX_SCREENSHOTS_PER_PAGE = 12;
 
-class GroupEventAttachments extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      deletedAttachments: [],
-    };
-  }
-
-  getActiveAttachmentsTab() {
-    const {location} = this.props;
+function useActiveAttachmentsTab() {
+  const location = useLocation();
 
-    const types = decodeList(location.query.types);
-    if (types.length === 0) {
-      return EventAttachmentFilter.ALL;
-    }
-    if (types[0] === SCREENSHOT_TYPE) {
-      return EventAttachmentFilter.SCREENSHOTS;
-    }
-    if (xor(crashReportTypes, types).length === 0) {
-      return EventAttachmentFilter.CRASH_REPORTS;
-    }
+  const types = decodeList(location.query.types);
+  if (types.length === 0) {
     return EventAttachmentFilter.ALL;
   }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {params, location} = this.props;
-
-    if (this.getActiveAttachmentsTab() === EventAttachmentFilter.SCREENSHOTS) {
-      return [
-        [
-          'eventAttachments',
-          `/organizations/${params.orgId}/issues/${params.groupId}/attachments/`,
-          {
-            query: {
-              ...location.query,
-              types: undefined, // need to explicitly set this to undefined because AsyncComponent adds location query back into the params
-              screenshot: 1,
-              per_page: MAX_SCREENSHOTS_PER_PAGE,
-            },
-          },
-        ],
-      ];
-    }
-
-    return [
-      [
-        'eventAttachments',
-        `/organizations/${params.orgId}/issues/${params.groupId}/attachments/`,
-        {
-          query: {
-            ...pick(location.query, ['cursor', 'environment', 'types']),
-            per_page: 50,
-          },
-        },
-      ],
-    ];
+  if (types[0] === SCREENSHOT_TYPE) {
+    return EventAttachmentFilter.SCREENSHOTS;
   }
+  if (xor(crashReportTypes, types).length === 0) {
```

Comment:
ah yes a xor

Reply:
You know things are serious when you pull out the xor